surface/include/pcl/surface/impl/marching_cubes.hpp

```diff
@@ -167,20 +148,14 @@ pcl::MarchingCubes<PointNT>::createSurface (std::vector<float> &leaf_node,
     interpolateEdge (p[3], p[7], leaf_node[3], leaf_node[7], vertex_list[11]);
 
   // Create the triangle
-  for (int i = 0; triTable[cubeindex][i] != -1; i+=3)
+  for (int i = 0; triTable[cubeindex][i] != -1; i += 3)
   {
-    PointNT p1,p2,p3;
-    p1.x = vertex_list[triTable[cubeindex][i  ]][0];
-    p1.y = vertex_list[triTable[cubeindex][i  ]][1];
-    p1.z = vertex_list[triTable[cubeindex][i  ]][2];
+    PointNT p1, p2, p3;
+    p1.getVector3fMap () = vertex_list[triTable[cubeindex][i]];
     cloud.push_back (p1);
-    p2.x = vertex_list[triTable[cubeindex][i+1]][0];
-    p2.y = vertex_list[triTable[cubeindex][i+1]][1];
-    p2.z = vertex_list[triTable[cubeindex][i+1]][2];
+    p2.getVector3fMap () = vertex_list[triTable[cubeindex][i+1]];
     cloud.push_back (p2);
-    p3.x = vertex_list[triTable[cubeindex][i+2]][0];
-    p3.y = vertex_list[triTable[cubeindex][i+2]][1];
-    p3.z = vertex_list[triTable[cubeindex][i+2]][2];
+    p3.getVector3fMap () = vertex_list[triTable[cubeindex][i+2]];
     cloud.push_back (p3);
```

Comment:
I just noticed that this cloud is only being populated with XYZ info. Why exactly are we using PointNT?